openedx/core/djangoapps/credit/tests/test_api.py

```diff
@@ -514,6 +528,34 @@ def test_set_credit_requirement_status(self, mode):
     @ddt.data(
         *CourseMode.CREDIT_ELIGIBLE_MODES
     )
+    def test_set_credit_requirement_status_satisfied_to_failed(self, mode):
+
+        """
+        Test that if credit requirment status is set to `satisfied`, it
+        can not not be changed to `failed`
+        """
+        credit_course = self.add_credit_course()
+
+        # Enroll user and verify enrollment.
+        self.enroll(self.user, self.course_key, mode)
+        self.assertTrue(CourseEnrollment.is_enrolled(self.user, self.course_key))
+        self.assertTrue(CourseEnrollment.enrollment_mode_for_user(self.user, self.course_key), (mode, True))
+
+        # set requirements for course to earn credit
+        course_requirements = self._set_credit_course_requirements()
+
+        api.set_credit_requirement_status(self.user, self.course_key, "grade", "grade", status="satisfied")
+        self.assert_grade_requirement_status('satisfied', 0)
+
+        # try to set status to `failed`
+        api.set_credit_requirement_status(self.user, self.course_key, "grade", "grade", status="failed")
+
+        # status should not be changed to `failed`, rather should maintain already set status `satisfied`
+        self.assert_grade_requirement_status('satisfied', 0)
+
+    @ddt.data(
```

Comment:
I think it should be added here as there are multiple modes from which credit can be earned and we need to run tests for all those modes.